actix-http/src/header/shared/httpdate.rs

```diff
@@ -1,59 +1,50 @@
 use std::fmt::{self, Display};
 use std::io::Write;
 use std::str::FromStr;
-use std::time::{Duration, SystemTime, UNIX_EPOCH};
+use std::time::{SystemTime, UNIX_EPOCH};
 
 use bytes::{buf::BufMutExt, BytesMut};
 use http::header::{HeaderValue, InvalidHeaderValue};
+use time::PrimitiveDateTime;
 
 use crate::error::ParseError;
 use crate::header::IntoHeaderValue;
 
 /// A timestamp with HTTP formatting and parsing
 #[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
-pub struct HttpDate(time::Tm);
+pub struct HttpDate(time::PrimitiveDateTime);
 
 impl FromStr for HttpDate {
     type Err = ParseError;
 
     fn from_str(s: &str) -> Result<HttpDate, ParseError> {
-        match time::strptime(s, "%a, %d %b %Y %T %Z")
-            .or_else(|_| time::strptime(s, "%A, %d-%b-%y %T %Z"))
-            .or_else(|_| time::strptime(s, "%c"))
+        match PrimitiveDateTime::parse(s, "%a, %d %b %Y %H:%M:%S")
```

Comment:
this may fail as well

Reply:
This is similar to the other bit of parsing logic. Without the %Z timezone format pattern, the timezone is simply ignored and assumed to be UTC.